src/commands/goToDefinition.ts

```diff
@@ -0,0 +1,212 @@
+  if (vscode.window.activeNotebookEditor) {
+    // We're in a notebook, so we're navigating back to code
+
+    // Find the cell that we're in
+    const notebookEditor = vscode.window.activeNotebookEditor;
+
+    const cell = vscode.window.activeNotebookEditor.notebook.cellAt(
+      notebookEditor.selections[0].start
+    );
+    if (!cell) {
+      return;
+    }
+
+    let moduleName = findCellModuleName(cell);
+
+    // We don't have a cell-local export, find the default one
+    if (!moduleName) {
+      // Get text from all cells in the document
+      const allCellsText = vscode.window.activeNotebookEditor.notebook
+        .getCells()
+        .map((cell) => cell.document.getText())
+        .join("");
+
+      // Search for the `#| default_exp` comment
+      const defaultExpMatch = allCellsText.match(/#\|\s*default_exp\s+(\w+)/);
+
+      if (defaultExpMatch) {
+        // Read the path from the default_exp comment
+        const [_, name] = defaultExpMatch;
+        moduleName = name;
+      }
+    }
+
+    // If the module name is still undefined, we can't do anything
+    // This happens if the user is in a cell that doesn't have an export comment
+    // and there is no default_exp comment in the document
+    if (!moduleName) {
+      return;
+    }
+
+    // Get the root of the workspace
+    const workspaceRoot = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+    if (!workspaceRoot) {
+      return;
+    }
+
+    // Open the settings.ini file
+    const settingsPath = path.join(workspaceRoot, "settings.ini");
+
+    // Get the text from the settings.ini file
+    const settingsText = await vscode.workspace.fs.readFile(
+      vscode.Uri.file(settingsPath)
+    );
+
+    // lib_ath regex
+    const libPathPattern = /lib_path\s*=\s*([^;\s]+)/;
+
+    // Find the lib_path in the settings.ini file
+    const libPathMatch = settingsText.toString().match(libPathPattern);
+    if (!libPathMatch) {
+      return;
+    }
+
+    // Read the path from the lib_path comment
+    const [_, libPath] = libPathMatch;
+
+    // Get the full path to the module
+    const modulePath = path.join(workspaceRoot, libPath, `${moduleName}.py`);
+
+    // Open the module
+    const moduleUri = vscode.Uri.file(modulePath);
+    await vscode.commands.executeCommand(
+      "vscode.openWith",
+      moduleUri,
+      "python"
+    );
+
+    // Get the editor for the module
+    const activeEditor = vscode.window.activeTextEditor;
+    if (!activeEditor) {
+      return;
+    }
+
+    // Search for the cell text in the activeEditor
+    const cellText = cell.document.getText();
+
+    // Remove the first line from the cellText - remove the characters until /n
+    const cellTextWithoutFirstLine = cellText.substring(
+      cellText.indexOf("\n") + 1
+    );
+
+    // Get the location of the text in the document
+    const cellTextLocation = activeEditor.document
+      .getText()
+      .indexOf(cellTextWithoutFirstLine);
```

Comment:
Use comments instead of searching for code?  I think that could be a better approach, you want to throw a warning if the nbs are not in sync though.  Perhaps leave that for P2